experimenter/tests/integration/nimbus/android/conftest.py

```diff
@@ -0,0 +1,154 @@
+        run_nimbus_cli_command(command)
+        time.sleep(10)
+
+    return _
+
+
+@pytest.fixture
+def gradlewbuild_log(pytestconfig, tmpdir):
+    gradlewbuild_log = f"{tmpdir.join('gradlewbuild.log')}"
+    pytestconfig._gradlewbuild_log = gradlewbuild_log
+    yield gradlewbuild_log
+
+
+@pytest.fixture
+def gradlewbuild(gradlewbuild_log):
+    yield GradlewBuild(gradlewbuild_log)
+
+
+@pytest.fixture(name="ping_server", autouse=True, scope="session")
+def fixture_ping_server():
+    path = next(iter(here.glob("**/android/ping_server")))
+    process = start_process(path, ["python", "ping_server.py"])
+    yield "http://localhost:5000"
+    process.terminate()
+
+
+@pytest.fixture(name="delete_telemetry_pings")
+def fixture_delete_telemetry_pings(ping_server):
+    def runner():
+        requests.delete(f"{ping_server}/pings")
+
+    return runner
+
+
+@pytest.fixture(name="check_ping_for_experiment")
+def fixture_check_ping_for_experiment(experiment_slug, ping_server, open_app):
+    def _check_ping_for_experiment(branch=None, experiment=experiment_slug, reason=None):
+        model = TelemetryModel(branch=branch, experiment=experiment)
+
+        timeout = time.time() + 60
+        while time.time() < timeout:
+            data = requests.get(f"{ping_server}/pings").json()
+            events = []
+            for item in data:
+                event_items = item.get("events")
+                if event_items:
+                    for event in event_items:
+                        if (
+                            "category" in event
+                            and "nimbus_events" in event["category"]
+                            and "extra" in event
+                            and "branch" in event["extra"]
+                        ):
+                            events.append(event)
+            for event in events:
+                event_name = event.get("name")
+                if (reason == "enrollment" and event_name == "enrollment") or (
+                    reason == "unenrollment"
+                    and event_name in ["unenrollment", "disqualification"]
+                ):
+                    telemetry_model = TelemetryModel(
+                        branch=event["extra"]["branch"],
+                        experiment=event["extra"]["experiment"],
+                    )
+                    if model == telemetry_model:
+                        return True
+            time.sleep(5)
+            open_app()  # Open app in cycles to trigger telemetry
+        return False
+
+    return _check_ping_for_experiment
+
+
+@pytest.fixture(name="set_experiment_test_name", autouse=True, scope="session")
+def fixture_set_experiment_test_name():
+    os.environ["EXP_NAME"] = "fenix"
+
+
+@pytest.fixture(name="setup_experiment")
+def fixture_setup_experiment(
+    run_nimbus_cli_command,
+    delete_telemetry_pings,
+):
+    def _():
+        experiment_json = next(iter(here.glob("**/fixtures/experiment.json")))
```

Comment:
The problem is starting the fenix app with automation. Espresso won't allow you to just start the app cleanly. I'd have to use Appium, and with Appium it takes over the running adb instance, so I'd have to do some weird juggling of ADB and who controls the emulator at a certain time. Using `nimbus-cli`, it still gives us the telemetry we need in the end.

Reply:
Okay I see, still another way to do this is to stand up experimenter, create experiment, fetch the json, and inject it with the CLI the way you're doing.  If we hardcode the recipe in here, then we could break something in Experimenter and this test would pass.  We should use the fact that experimenter is present here to ensure that Experimenter and Fenix integration stays intact.  We can do it as a followup since this is already passing self contained.